os/mm/mm_heap/mm_getheap.c

```diff
@@ -71,24 +71,20 @@ typedef struct app_heap_s {
 
 static dq_queue_t app_heap_q;
 
-#ifdef CONFIG_MM_PARTITION_HEAP
-extern struct mm_heap_s g_pheap;
-#endif
-
 #endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
-#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+#if defined(CONFIG_APP_BINARY_SEPARATION)
 void mm_initialize_app_heap()
 {
 	dq_init(&app_heap_q);
 }
 
 void mm_add_app_heap_list(struct mm_heap_s *heap, char *app_name)
 {
-	app_heap_s *node = (app_heap_s *)kmm_malloc(sizeof(app_heap_s));
+	app_heap_s *node = (app_heap_s *)malloc(sizeof(app_heap_s));
```

Comment:
If you use malloc here from where this memory is allocated? user heap of currently running application?